index-import.cpp

```diff
@@ -210,45 +195,89 @@ static bool cloneRecords(StringRef recordsDirectory,
 
     std::error_code failed;
     if (status->type() == fs::file_type::directory_file) {
-      if (not fs::exists(outputPath)) {
-        failed = fs::create_directory(outputPath);
+      failed = fs::create_directory(outputPath);
+      if (failed && failed != std::errc::file_exists) {
+        success = false;
+        errs() << "error: " << strerror(errno) << "\n"
+               << "\tcould not create directory: " << outputPath << "\n";
       }
     } else if (status->type() == fs::file_type::regular_file) {
       // Two record files of the same name are guaranteed to have the same
       // contents (because they include a hash of their contents in their
       // file name). If the destination record file already exists, it
       // doesn't need to be cloned or copied.
-      if (not fs::exists(outputPath)) {
-        failed = fs::copy_file(inputPath, outputPath);
+      failed = fs::copy_file(inputPath, outputPath);
```

Comment:
Perhaps the comment needs rewording. The `fs::exists` check avoids performing a redundant copy operation, as a performance optimization. Thanks to the content based hash, if `outputPath` exists, then this record file can be skipped because it already exists and has the same contents.

Reply:
No, I thought `copy_file()` failed when the destination file already existed. I tested and this is not the case... but see my comments on the other RB... we might want both.

Reply:
I think `fs::copy_file` is part of the problem, the use cases it supports are more narrow. In #15, `fs::copy_file` is no longer used, but it's intended to be temporary. Maybe the right way forward is to use `copyfile` directly. If ever building for linux, then we'd have to add portable fallbacks where cloning doesn't exist.

Reply:
I'd prefer the "don't copy if exists" logic be explicit, rather implicit behind a `copyfile` flag. In my testing of incremental use (where the output directory already exists), then the two approaches have the same perf.

Reply:
reverted to explicit logic